### PR TITLE
ARTEMIS-3751 Do not override large message size

### PR DESCRIPTION
**Pull request by iliya-gr**
It is possible to receive a compressed message from the client as regular message. Such a message will already contain correct body size, that takes compression into account.

### CODE REVIEW

File: tests/integration-tests/src/test/java/org/apache/activemq/artemis/tests/integration/amqp/JMSLargeMessageTest.java [hunk 1]
```diff
@@ -0,0 +1,192 @@
...
+      MessageConsumer consumer = session.createConsumer(queue);
+
+      connection.start();
+
+      // What we want is to get a message witch size is larger than client's minLargeMessageSize (200 kibs here) while compressed the size must be lower than that
+      // and at same time it must be greater than server minLargeMessageSize (which is 100 kibs by default)
+      byte[] data = new byte[1024 * 300];
+
+      // We don't want the data to be too random, 42 seems to be a good random seed
+      new DeflateGenerator(new Random(42), 2.0, 2.0).generate(data, 2);
```
Comment (clebertsuconic) [inline tests/integration-tests/src/test/java/org/apache/activemq/artemis/tests/integration/amqp/JMSLargeMessageTest.java:53]: I'm trying to understand why you're using Deflate directly? why not use the compressed from the ConnectionFActory directly and why you're compressing it yourself?

Comment (clebertsuconic) [reply tests/integration-tests/src/test/java/org/apache/activemq/artemis/tests/integration/amqp/JMSLargeMessageTest.java:53]: is this just to generate Random Data?

Comment (iliya-gr) [reply tests/integration-tests/src/test/java/org/apache/activemq/artemis/tests/integration/amqp/JMSLargeMessageTest.java:53]: It's only for generating random data. Assert is to ensure that it matches criteria: `server large message size < compressed size < client large message size`

Comment (iliya-gr) [reply tests/integration-tests/src/test/java/org/apache/activemq/artemis/tests/integration/amqp/JMSLargeMessageTest.java:53]: Assert doesn't change data. Deflate in assert is just for checking parameters, if someone decides to change them.

Comment (clebertsuconic) [reply tests/integration-tests/src/test/java/org/apache/activemq/artemis/tests/integration/amqp/JMSLargeMessageTest.java:53]: >> Deflate in assert is just for checking parameters, if someone decides to change them. <<

I'm not sure what you meant by this statement. Can you speak in java? :)

Comment (iliya-gr) [reply tests/integration-tests/src/test/java/org/apache/activemq/artemis/tests/integration/amqp/JMSLargeMessageTest.java:53]: assertCompressionSize - doesn't modify input data. It's for checking that DeflateGenerator has generated correct data:

```java
deflate(data).length > storageManager.getMaxRecordSize() && deflate(data).length < serverLocator.minLargeMessageSize
```

Comment (iliya-gr) [reply tests/integration-tests/src/test/java/org/apache/activemq/artemis/tests/integration/amqp/JMSLargeMessageTest.java:53]: DeflateGenerator - is just compressible data generator 
generate(data, 2) - generate compressible data with compression ratio 2 and write it in data

Comment (clebertsuconic) [reply tests/integration-tests/src/test/java/org/apache/activemq/artemis/tests/integration/amqp/JMSLargeMessageTest.java:53]: I am merging this with a small adjustment...

the package where you added the test was meant for AMQP tests only... I have moved  your test within LargeMessageCompressTest...


I am keeping you as the author on the commit...

Comment (clebertsuconic) [reply tests/integration-tests/src/test/java/org/apache/activemq/artemis/tests/integration/amqp/JMSLargeMessageTest.java:53]: as I said on the JIRA the issue is that the message was sent as a regular message, and before writing we check the journal-buffer-size on the server's.. if the buffer-size < message body size, we convert it to large message on the server's side.. this is kind of a protection.

it is a bit of an edge case as you were really supposed to send with minLargeMessageSize = journal-buffer-size... 


I even had users requesting this to be enforced but I could never find a way to do that without breaking the protocol.


I will merge this though, but you should really use minLargeMessageSize == journal-buffer-size.

**Pull request closed without merging**